poetry/console/commands/shell.py

```diff
@@ -0,0 +1,55 @@
+from os import environ
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        from poetry.config import Config
+
+        # Check if it's already activated or doesn't exist and won't be created
+        if "POETRY_ACTIVE" in environ or not self.venv.is_venv():
```

Comment:
I would prefer explicit `if environ.get("POETRY_ACTIVE") == "1"` check (maybe split it into two parts then). That way you control for exactly what you are setting up later. Assuming some weird case when for whatever reason someone would have an env var POETRY_ACTIVE=0 set, this could work other then intended

Reply:
Fair enough.